bindings/netstandard/ElectionGuard/ElectionGuard.Encryption/PrecomputeAPI.cs

```diff
@@ -142,19 +152,43 @@ public PrecomputeStatus GetStatus()
             return currentStatus;
         }
 
+        /// <summary>
+        /// Gets the progress of the precompute
+        /// </summary>
+        /// <param name="count">count of the buffer entries</param>
+        /// <param name="queue_size">max size of the buffer queue</param>
+        public void GetProgress(out int count, out int queue_size)
+        {
+            NativePrecomputeBuffers.Status(out count, out queue_size);
+        }
+
         /// <summary>
         /// Starts the precompute process by creating a new thread to run the process
         /// </summary>
         /// <param name="maxexp">The max exponentiation to be calculated</param>
         /// <param name="token">CancelationToken that can be used to start the process</param>
+        [Obsolete]
         public void StartPrecomputeAsync(long maxexp, CancellationToken token)
         {
-            cancelTokenSource = CancellationTokenSource.CreateLinkedTokenSource(cancelToken);
-            cancelToken = token;
             currentStatus.CurrentState = PrecomputeState.Running;
-            delay = maxexp;
+        }
+
+        /// <summary>
+        /// Starts the precompute process by creating a new thread to run the process
+        /// </summary>
+        /// <param name="publicKey">The max exponentiation to be calculated</param>
+        /// <param name="token">CancelationToken that can be used to start the process</param>
+        /// <param name="buffers">The maximum number of buffers to precompute</param>
+        public void StartPrecomputeAsync(ElementModP publicKey, CancellationToken token, int buffers = 0)
+        {
+            currentStatus.CurrentState = PrecomputeState.Running;
+            max_buffers = buffers;
+            elgamalPublicKey = publicKey;
+
             workerThread = new Thread(WorkerMethod);
+            workerThread.Name = "Precompute Worker Thread";
             workerThread.Start();
+            Thread.Sleep(500);

```

Comment:
Is this Thread.Sleep is really necessary?  If so, it should at least have a comment.  However, it makes me nervous as it usually indicates a race condition exists somewhere.

Reply:
This Sleep is so that there is not a race condition where the start is called and then stop is called before it gets a chance to even get started.  This is meant to protect the callers from getting into a race condition.

Reply:
Removed.